### PR TITLE
Cohort2 questions

### PR DESCRIPTION
**Pull request by kevin-m-kent**
adding cohort2 questions that arose during discussions to QandA folder. Links are to points in video where questions were asked and discussed.

### CODE REVIEW

**Comment by MayaGans** [comment]
I think we discussed this a little on slack but I think we should add the chapter specific questions to the chapter files (ie chapter 2 questions here: https://github.com/kevin-m-kent/bookclub-Advanced_R/blob/main/QandA/02-names_and_values.Rmd) 

That said I do also like the idea of a `guiding questions` markdown to help maybe we can add that to the root dir? 

What do you think?

**Comment by tanho63** [comment]
I like the idea of a guiding questions markdown! I think it's similar to the outline/quiz sections in the early chapters of the book. Wonder if it's still necessary longer-term.

I had previously noted that the README has a few conflicting changes stored in it (looks like it's trying to restore an older edition of the README).

File: QandA/Cohort-2-Questions.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- For exercise 2, section 2.5.1, why do I get one less copy when running code?
+    - It could be a difference between R and Rstudio. You also may need to switch your environment to something other than global. 
+- How does R allocate memory for different length character strings? 
+    - Seems to be allocating bytes in increments of ~8 characters (https://youtu.be/pCiNj2JRK50?t=988) 
+- Why don’t lists with repeated values that aren’t characters point to the same object, like character vectors do? (https://youtu.be/pCiNj2JRK50?t=2916) 
+- If you have to run garbage collector, are you doing something wrong with your code? (https://youtu.be/pCiNj2JRK50?t=1686) 
+- How is removing objects dealt with in the garbage collection process? (https://youtu.be/pCiNj2JRK50?t=1811) 
+- When might you want to use local? (https://youtu.be/pCiNj2JRK50?t=1869) 
+- Why does the example in section 2.5.1 make 3 copies when its a data frame? (https://youtu.be/pCiNj2JRK50?)t=2096) 
+- Where can I learn more about copy behavior with primitive functions? (https://youtu.be/pCiNj2JRK50?t=2587)
```
Comment (MayaGans) [inline QandA/Cohort-2-Questions.md:11]: You can add these to `02-names_and_values.Rmd` where I made `<h2>` headers for each section and annotate questions and answers as:

```
## 2.5.1.2 Exercise {-}

:::question
Why do I get one less copy when running code?
:::

:::TODO
:::
```

[replacing TODO with the answer if answered!

...That said I think this specific question is already answered in the text but that should provide a template 😄

File: QandA/Cohort-2-Questions.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  )
+​
+df %>%
+  mutate(
+    date = map2(enter_date, exit_date, seq.Date, by = "day")
+  ) %>%
+  unnest(date)
+
+  ```
+
```
Comment (MayaGans) [inline QandA/Cohort-2-Questions.md:39]: As above, please add these into `03-vectors.Rmd`

File: README.md [hunk 3]
```diff
@@ -30,16 +31,13 @@ This repo contains 3 main folders:
...
-  Week 2
-
-- What is the difference between an object’s name and its values?
-- When you modify a vector, when does a copy occur? Are there cases when a copy isn’t created?
-- How do you check how much space and object occupies? What kinds of objects occupy the most/least space?
-- When does R’s garbage collection usually run? How do you manually clean up memory?
+- Why are there six methods for subsetting atomic vectors? Are there different use cases or purposes for each? 
+- What are the similarities and differences between subsetting vectors, lists, dataframes, and tibbles? 
+- What is the difference between simplifying and preserving when subsetting? 
+- What are the most common use cases for subsetting?
```
Comment (MayaGans) [inline README.md:40]: I like the idea of making a guiding questions its own markdown OR I can make a div 

```
:::guidingq
:::
```

So we start each chapter of the bookdown with these questions? I kind of love that idea.... Thoughts?

File: README.md [hunk 4]
```diff
@@ -74,19 +73,28 @@ This repo contains 3 main folders:
...
+<details>
+  <summary> Guiding Questions </summary>
+
+- What is the difference between an object’s name and its values?
+- When you modify a vector, when does a copy occur? Are there cases when a copy isn’t created?
+- How do you check how much space and object occupies? What kinds of objects occupy the most/least space?
+- When does R’s garbage collection usually run? How do you manually clean up memory?
+
+
+</details>
```
Comment (MayaGans) [inline README.md:92]: Comment above as per a home for guiding questions

**Comment by MayaGans** [comment]
also looks like you'll need to update the readme and resolve conflicts before we can merge 😄

**Comment by kevin-m-kent** [comment]
Sorry I've been slow with responding here! I can try to get to the rmd Q&A today or tomorrow. Thanks for your comments

File: README.md (hunk 3 above)
Comment (tanho63) [reply README.md:40]: I like this, I know Jon was working on similar. Wonder where to fit it into the READMEs.

**Comment by tanho63** [comment]
Pinging a followup as I go through and clean up some other things going on here :)

File: README.md (hunk 3 above)
Comment (kevin-m-kent) [reply README.md:40]: I like this idea as well - starting each chapter with guiding questions.

**Comment by kevin-m-kent** [comment]
It sounds like everyone thinks this could go in the bookdown at the beginning of each chapter?

**Comment by tanho63** [comment]
Yeah, I think so @kevin-m-kent . 

@jonthegeek did you have learning objectives put together in a slide deck for Adv R? Thought I saw you did at one point

**Comment by tanho63** [comment]
Just closing this as a stale PR, let's collab on the issue in #190 to figure out how to do it best!

**Pull request closed without merging**